File: test-suite-python/src/test/python/micronaut/views/docs/fieldset/AuthorRepository.py
from jakarta.inject import Singleton

from .Author import Author


@Singleton
class AuthorRepository:

    def find_all(self) -> list[Author]:
        return [
            Author(1, "Kishori Sharan"),
            Author(2, "Peter Späth"),
            Author(3, "Sam Newman"),
        ]

File: test-suite-kotlin/src/test/kotlin/io/micronaut/views/docs/fieldset/AuthorRepository.kt
package io.micronaut.views.docs.fieldset

import jakarta.inject.Singleton

@Singleton
class AuthorRepository {

    fun findAll(): List<Author> = listOf(
        Author(1L, "Kishori Sharan"),
        Author(2L, "Peter Späth"),
        Author(3L, "Sam Newman"))
}
